### PR TITLE
[Bug] Fix some issues to make rag.ipynb work

### PR DESCRIPTION
**Pull request by kevin85421**
The notebook `rag.ipynb` doesn't work for some reason. This PR fixes those issues to improve UX. See my comments for this PR for more details.

### CODE REVIEW

File: notebooks/rag.ipynb [hunk 1]
```diff
@@ -198,10 +198,11 @@
    ],
    "source": [
     "# Credentials\n",
+    "# Comment out OpenAI related environment variables if you use Anyscale endpoints, and vice versa.\n",
```
Comment (kevin85421) [inline notebooks/rag.ipynb:201]: If we don't comment out OpenAI env variables, this code block will fail.

File: notebooks/rag.ipynb [hunk 2]
```diff
@@ -1045,105 +1048,20 @@
...
-    }
-   ],
+   "outputs": [],
    "source": [
     "%%bash\n",
-    "# Drop table and load index\n",
+    "# Drop the existing `document` table and create a new one with the schema to store embeddings. \n",
     "psql \"$DB_CONNECTION_STRING\" -c \"DROP TABLE IF EXISTS document;\"  # drop\n",
     "sudo -u postgres psql -f $MIGRATION_FP  # create\n",
-    "psql \"$DB_CONNECTION_STRING\" -f $SQL_DUMP_FP  # load\n",
```
Comment (kevin85421) [inline notebooks/rag.ipynb:1145]: `$SQL_DUMP_FP` will only be created after the embeddings are written into Postgres in the following step, as shown below. Hence, we don't have the file `$SQL_DUMP_FP` at this moment.

```sh
sudo -u postgres pg_dump -c > $SQL_DUMP_FP  # save
```
Screen Shot 2024-03-26 at 12 25 00 AM

File: notebooks/rag.ipynb [hunk 3]
```diff
@@ -1222,7 +1140,11 @@
     "    batch_size=128,\n",
     "    num_cpus=1,\n",
     "    compute=ActorPoolStrategy(size=6),\n",
-    ").count()"
+    ").materialize()\n",
```
Comment (kevin85421) [inline notebooks/rag.ipynb:1143]: `.count()` will not execute the function `StoreResults` indefinitely. I referred to [this blog](https://www.anyscale.com/blog/rag-at-scale-10x-cheaper-embedding-computations-with-anyscale-and-pinecone), and used `materialize()` instead.

File: setup-pgvector.sh [hunk 4]
```diff
@@ -5,7 +5,7 @@ wget --quiet -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | sudo apt-
 sudo sh -c 'echo "deb http://apt.postgresql.org/pub/repos/apt/ $(lsb_release -cs)-pgdg main" >> /etc/apt/sources.list.d/pgdg.list'
 sudo apt update -y && sudo apt install -y postgresql postgresql-contrib
 # Install pgvector
-sudo apt install -y postgresql-server-dev-16
+sudo apt install -y postgresql-server-dev-all
```
Comment (kevin85421) [inline setup-pgvector.sh:8]: Screen Shot 2024-03-26 at 12 19 17 AM

**Comment by kevin85421** [comment]
cc @jjyao

**Comment by jjyao** [comment]
Can you update the PR title and description to be more descriptive?

**Comment by kevin85421** [comment]
> Can you update the PR title and description to be more descriptive?

Done. I just added a pointer in the PR description to my existing comments. Let me know if it is not enough.

**Comment by GokuMohandas** [comment]
Thanks @kevin85421 !

**Pull request merged**